modules/google-maps/src/utils.js

```diff
@@ -106,8 +106,14 @@ export function getViewState(map, overlay) {
   const nw = projection.fromContainerPixelToLatLng(nwContainerPx);
   const nwDivPx = projection.fromLatLngToDivPixel(nw);
 
+  // checking the number of maps visible on container
+  var mapCount = width / projection.getWorldWidth()
+  if (mapCount >= 1) {
+    nwDivPx.x = nwDivPx.x - projection.getWorldWidth()
```

Comment:
What happens when `mapCount > 2`? Which copy does `fromLatLngToDivPixel` project to?

Reply:
If the change above were omitted, and the change below (compute scale from Y axis, not X) were kept, what is the result? deckgl would draw on the leftmost visible map? A demo might be helpful – trying to be sure I understand what each of these changes is doing.

Reply:
@Pessimistress the above code just check that how many maps are visible on the container screen, so in case of zoom greater than 1 mapCount is always less then 1. It is for handling layers on zoom level 1. The fromLatLngToDivPixel projects to the map which is more visible on container

@donmccurdy, yes you are right if above code is omitted and scale computation is done using Y axis, then deck layers are visible on zoom level 2. It is just to handle the zoom level 1. It will handle the left property of canvas, so that it remains on the map which is more visible.

Reply:
The zoom levels are not coupled to mapCount in the way you appear to be assuming – the resolution of the screen and user agent zooming will both affect the layout as well. On the lowest zoom level you might get anywhere from 1 visible map to many visible maps:

<img width="996" alt="Screen Shot 2019-09-24 at 8 35 39 AM" src="https://user-images.githubusercontent.com/1848368/65526786-7af00b00-dea6-11e9-9617-2c351649b20e.png">

^That's an extreme case, of course, but the logic above — shifting the canvas by a single world width — does not look robust enough to me. If the change to `scale` alone improves things, it might be better to do that first and consider which map to render over with another PR? Or if you want to try to implement it here, I would suggest trying to use the centermost map, and assuming there could be any number of maps.